### PR TITLE
Allow pattern matching on key

### PR DESCRIPTION
**Pull request by pderaaij**
This PR makes it possible to create rules that does pattern matching on the key of an object.

For example, for the key of a path you can create the following rule:

>        {
>            "name":"path-item-lowercased-hyphens",
>            "object": "paths",
>            "enabled": true,
>            "description": "path items must be lowercase and separated with hyphens",
>            "pattern": { "property": "$key", "split": "/", "value": "^{[^}]*}$|^([a-z-\/]*)$" }
>        }

This relates to #214

### CODE REVIEW

**Comment by pderaaij** [comment]
Unsure what I can do about the breaking builds. Seems `circleci` related.

File: lib/linter.js [hunk 1]
```diff
@@ -151,7 +151,7 @@ const lint = (objectName, object, key, options = {}) => {
             }
             if (rule.pattern) {
                 const { omit, property, split, value } = rule.pattern;
-                const target = object[property]
+                let target = (property === '$key') ? key : object[property];
```
Comment (MikeRalphson) [inline lib/linter.js]: I think this can still be a `const` as there is only one assignment from the RHS of that expression?

**Comment by MikeRalphson** [comment]
@pderaaij I think you're right. Also patch LGTM for the approach.

File: lib/linter.js (hunk 1 above)
Comment (pderaaij) [reply lib/linter.js]: My mistake, fixed and squashed the commits

**Comment by bottoy** [comment]
I have been testing this new rule together with the linter update (https://github.com/Mermade/oas-kit/commit/310fd30b5da70a32dde8bcb2a4fe1e83d12fdf63) for the 'responses' object. 

'{
            "name": "response_number",
            "object": "responses",
            "enabled": true,
            "description": "The response needs to be a number value",
            "pattern":  { "property": "$key", "value": "^([0-9])*$"}
}'

I think code in speccy - linter.js should be 
'const target = (property === '$key') ? Object.keys(object)[0] : object[property];'

With 'response' object only I do not get the key value, but the string referring to the pattern (eg. response)

generic question:
* When the OAS has an object to validate the key (eg response key, path key, ...) you do it from the parent object (paths, responses) or from within each item (pathItem, response) - or we allow the both

Cheers,
Bram

**Comment by MikeRalphson** [comment]
Hi @bottoy I think @pderaaij has just fixed this over in `oas-kit` in https://github.com/Mermade/oas-kit/pull/134

**Pull request merged**